vaadin-maps-leaflet-flow-demo/src/main/java/software/xdev/vaadin/maps/leaflet/flow/demo/LeafletView.java

```diff
@@ -1,113 +1,116 @@
-	private final LMarker markerAsia = new LMarker(49.675039, 12.162127);
-	private final LMarker markerGreek = new LMarker(49.675126, 12.161899);
-	private final LMarker markerBakery = new LMarker(49.674806, 12.160249);
-	private final LMarker markerSchnitzel = new LMarker(49.673800, 12.160113);
-	private final LMarker markerBratwurst = new LMarker(49.674907, 12.161937);
+	private final LDefaultComponentManagementRegistry reg;
+	private final LMap map;
+	private final HorizontalLayout hlButtons = new HorizontalLayout();
 	
 	public LeafletView()
 	{
-		this.initMapComponents();
-		
-		this.btnLunch.addClickListener(this::btnLunchClick);
-		final HorizontalLayout hlButtonContainer = new HorizontalLayout();
-		hlButtonContainer.setWidthFull();
-		hlButtonContainer.setJustifyContentMode(JustifyContentMode.BETWEEN);
-		hlButtonContainer.setPadding(false);
-		hlButtonContainer.setSpacing(false);
-		hlButtonContainer.add(
-			this.btnLunch,
-			new Button("Center on Caribbean", e -> this.map.centerAndZoom(
-				new LPoint(14.467727, -61.69703),
-				new LPoint(16.33426, -60.921676))),
-			new Button("Open dialog over map", ev ->
-			{
-				final Icon icoClose = VaadinIcon.CLOSE.create();
-				
-				final Dialog dialog = new Dialog(icoClose);
-				dialog.setWidth("50vw");
-				dialog.setHeight("50vh");
-				dialog.open();
-				
-				icoClose.addClickListener(iev -> dialog.close());
-			}));
-		this.add(this.map, hlButtonContainer);
+		this.setId(ID);
+		
+		this.reg = new LDefaultComponentManagementRegistry(this);
+		
+		final MapContainer mapContainer = new MapContainer(this.reg);
+		mapContainer.setSizeFull();
+		this.add(mapContainer);
 		this.setSizeFull();
-	}
-	
-	private void btnLunchClick(final ClickEvent<Button> event)
-	{
-		this.viewLunch = !this.viewLunch;
-		
-		final List<LComponent> normalComponents = Arrays.asList(this.markerRathaus, this.markerZob);
-		final List<LComponent> lunchComponents = Arrays.asList(
-			this.circleRange,
-			this.markerPizza,
-			this.markerKebab,
-			this.markerAsia,
-			this.markerGreek,
-			this.markerBakery,
-			this.markerSchnitzel,
-			this.markerBratwurst);
-		
-		this.map.setViewPoint(new LCenter(49.675126, 12.160733, this.viewLunch ? 16 : 17));
-		this.map.removeLComponents(this.viewLunch ? normalComponents : lunchComponents);
-		this.map.addLComponents(this.viewLunch ? lunchComponents : normalComponents);
-		
-		this.btnLunch.setText(this.viewLunch ? "Go back to the normal view" : "Where do XDEV employees go for lunch?");
-	}
-	
-	private void initMapComponents()
-	{
-		this.markerZob.setPopup("Central bus station");
 		
-		final LMarker markerXDev = new LMarker(49.675806677512824, 12.160990185846394);
-		final LIcon xDevLogo = new LIcon(
-			"""
+		this.map = mapContainer.getlMap();
+		
+		final LLatLng center = new LLatLng(this.reg, 49.675126, 12.160733);
+		this.map.setView(center, 17);
+		
+		final LTileLayer tlOSM = LTileLayer.createDefaultForOpenStreetMapTileServer(this.reg);
```

Comment:
So every component has to have a ``LComponentManagementRegistry`` now? Why? Could it not be only added to the components that need it and omitted for components that don't need it?

Reply:
> So every component has to have a LComponentManagementRegistry now? Why?

Every component can be reused or has certain methods that may be revoked on a later point of time. This is not possible with the current (v3) "fire and forget" mechanism because once the component is created in JS we can't access it anymore from Java side. So the components need to be tracked somewhere and this is achieved using the ``LComponentManagementRegistry``.

> Could it not be only added to the components that need it and omitted for components that don't need it?

In theory it might be omitable from unmodifiable components (e.g. ``LPoint``) however there would be a different system required to handle these components which I think would be more confusing and make the whole system more complex. This would also not allow to reuse the created component.